### ISSUE TITLE
Fingerprint - chiper not initialized

### ISSUE THREAD

**Issue by hajimammada** [Kunzisoft/KeePassDX#1978]
I activated Biometric from KeepassDX settings. When I write my password and click Device unlock link it doesnt pop up fingerprint entering window. Instead, it retrieves an information "Please write your password first then click this button" (But I did that exact same way.

I solved this bug for myself with a different workaround:

First, I activated "Device credential unlocking" to use my device PIN. 
Then I came back to login place.
I write my password to the field and clicked Device unlock link.
I didn't enter any PIN and came back to the KeepassXC settings to deactivate "Device credential unlocking". Then I activated "Biometric unlocking". After that I returned to login field and clicked Device unlocking link again. This time fingerprint pop up appeared and successfully added my fingerprint.

So if someone encountered these errors below, simply can use this way or find a similar workaround:

"Please first write password then click this (Device unlocking link) button"

"Chiper not initialized"

**KeePassDX:**

 - Version: [e.g. 2.5.0.0beta23]
 - Build: [e.g. Free]
 - Language: [e.g. French]

**Android:**

 - Device: Xiaomi POCO X3 Pro
 - Version: MIUI 13.0.3

**Comment by KeronCyst** [comment]
This didn't work for me, unfortunately; I keep getting `AdvancedUnlockManager not initialized` no matter what I try, so I keep having to type my master password in every time. Not sure of when this will be fixed, as I certainly don't have the know-how...

**Comment by hajimammada** [comment]
> This didn't work for me, unfortunately; I keep getting `AdvancedUnlockManager not initialized` no matter what I try, so I keep having to type my master password in every time. Not sure of when this will be fixed, as I certainly don't have the know-how...

if you try a similar way or the exact same way that I tried, you will solve this for yourself.
That is a bug related to Keystore and we only can cheat the Keystore this way. But I agree there must be fixed otherwise it is annoyable.

**Comment by sanesecurityguy** [comment]
> First, I activated "Device credential unlocking" to use my device PIN.
Then I came back to login place.
I write my password to the field and clicked Device unlock link.
I didn't enter any PIN and came back to the KeepassXC settings to deactivate "Device credential unlocking". Then I activated "Biometric unlocking". After that I returned to login field and clicked Device unlocking link again. This time fingerprint pop up appeared and successfully added my fingerprint.

@hajimammada, your solution did not work for me.

> This didn't work for me, unfortunately; I keep getting `AdvancedUnlockManager not initialized` no matter what I try, so I keep having to type my master password in every time. Not sure of when this will be fixed, as I certainly don't have the know-how...

@KeronCyst, I had the same problem and accidentally fixed it myself somehow. I don't know if this was really what fixed it for me, but I went into Settings > Security > Fingerprint, and from there I clicked on "Learn more". It's supposed to open a webpage, but it ended up crashing for me and sending me back to the Security settings page. After that, KeePassDX's Biometric unlocking started working for me, and I stopped getting both the `AdvancedUnlockManager not initialized` error and the "cipher _something-something_" (forgot what exactly) error.

I didn't even need to set up Biometric unlocking in KeePassDX again; it was already set up from before when I first ran into those two errors. So my guess is that this is a problem with the Android Keystore, not with KeePassDX.

**Comment by rafaelrosadev** [comment]
I noticed this issue on my phone recently, it's a Poco F6. I'm not sure if it has any relation, but my phone received the Android 15 update a few days ago. Every time I start typing the password for my .kdbx, a toast message appears on the screen with the following message: 'Not implemented (internal keystore code: -100 message: system/security/keystore...'

**Comment by imrahulsharmaa** [comment]
> I noticed this issue on my phone recently, it's a Poco F6. I'm not sure if it has any relation, but my phone received the Android 15 update a few days ago. Every time I start typing the password for my .kdbx, a toast message appears on the screen with the following message: 'Not implemented (internal keystore code: -100 message: system/security/keystore...'

Same for me also i was using matrixx custom rom but when i updated to latest version, now biometric unlock option isn't working.

**Comment by kongjun18** [comment]
I am using Xiaomi 14 and has an error 'Not implemented (internal keystore code: -100 message: system/security/keystore...' too.

**Comment by kur0** [comment]
Same here, using Xiaomi/POCO official ROM.
It was working until recently, but other Keepass android app is working fine with the Fingerprint unlock db

**Comment by rolicot** [comment]
I had the same problem on Motorola Edge 50 fusion and none of the solutions above worked for me. What worked was simply erasing all application data (should be equivalent to uninstall/reinstall) and trying again. It could have been related to restoring of app settings from the older phone.

**Comment by J-Jamet** [comment]
I made a quick fix for version 4.1.3, and I will do a complete refactoring in version 5.0.0. But it should work, or at least display a better error message if the Keystore is inaccessible.

**Comment by J-Jamet** [comment]
For clarification, just because biometrics works on other applications doesn't mean it will work with KeePassDX. Most other applications use an asymmetrical key system, whereas KeePassDX uses a symmetrical key with a level of security "strong". If you're using a custom ROM and your system has been upgraded, there's a good chance that this keystore locking mode has been poorly implemented by the API. For example, the certificate needed to use the correct security system may be missing. Refer to the error code provided and ask the ROM team.

**Comment by rndev15** [comment]
Hey @J-Jamet thank you for your work 
I have the same issue on my Xiaomi 14 that @kongjun18 described above
The problem started after HyperOS update several days ago

I tried your beta from this message https://github.com/Kunzisoft/KeePassDX/issues/2098#issuecomment-3172866518, but unfortunately issue still there

Could you please try to fix that (in 4.1.4)? I can help you with testing if needed

**Comment by J-Jamet** [comment]
Yes, I'm in the process of redoing this part. I've redone the exception system and I'm working on the flow system for displaying prompts. The problem is that there's deprecated code to maintain for older phones and there are lots of different cases to test.
